Doctest failure in 'What's new in 5.0?'

I tried enabling doctests in the 5.0 What's New to make sure all the examples work fine and ran into the following failure:

```
=================================== FAILURES ===================================
______________________________ [doctest] 5.0.rst _______________________________
091 (and enabled by default) to allow for redshift - dimensionless conversions.
092 To convert between redshift and other cosmological distance measures, e.g.
093 CMB temperature or comoving distance, the equivalency ``with_redshift`` is
094 also added.
095 
096     >>> import astropy.units as u
097     >>> import astropy.cosmology.units as cu
098     >>> z = 1100 * cu.redshift
099 
100     >>> z.to(u.dimensionless_unscaled)
UNEXPECTED EXCEPTION: UnitConversionError("'redshift' and '' (dimensionless) are not convertible")
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest 5.0.rst[11]>", line 1, in <module>
  File "/home/runner/work/astropy/astropy/.tox/py39-test-cov/lib/python3.9/site-packages/astropy/units/quantity.py", line 849, in to
    value = self._to_value(unit, equivalencies)
  File "/home/runner/work/astropy/astropy/.tox/py39-test-cov/lib/python3.9/site-packages/astropy/units/quantity.py", line 803, in _to_value
    return self.unit.to(unit, self.view(np.ndarray),
  File "/home/runner/work/astropy/astropy/.tox/py39-test-cov/lib/python3.9/site-packages/astropy/units/core.py", line 1135, in to
    return self._get_converter(Unit(other),
  File "/home/runner/work/astropy/astropy/.tox/py39-test-cov/lib/python3.9/site-packages/astropy/units/core.py", line 1066, in _get_converter
    raise exc
  File "/home/runner/work/astropy/astropy/.tox/py39-test-cov/lib/python3.9/site-packages/astropy/units/core.py", line 1051, in _get_converter
    return self._apply_equivalencies(
  File "/home/runner/work/astropy/astropy/.tox/py39-test-cov/lib/python3.9/site-packages/astropy/units/core.py", line 1027, in _apply_equivalencies
    raise UnitConversionError(
astropy.units.core.UnitConversionError: 'redshift' and '' (dimensionless) are not convertible
/home/runner/work/astropy/astropy/docs/whatsnew/5.0.rst:100: UnexpectedException
```

If the example is incorrect, it should be fixed in the v5.0.x branch directly. Otherwise, what could be going wrong?

See https://github.com/astropy/astropy/runs/4081948989?check_suite_focus=true for the full log

--- a/docs/units/equivalencies.rst
+++ b/docs/units/equivalencies.rst
@@ -641,21 +641,7 @@ Examples
 
 .. EXAMPLE START: Using Equivalencies in Larger Pieces of Code
 
-To enable radians to be treated as a dimensionless unit with the
-:func:`~astropy.units.set_enabled_equivalencies` function:
-
-  >>> u.set_enabled_equivalencies(u.dimensionless_angles())
-  <astropy.units.core._UnitContext object at ...>
-  >>> u.deg.to('')  # doctest: +FLOAT_CMP
-  0.017453292519943295
-
-Here, any list of equivalencies could be used, or you could add, for example,
-:func:`~astropy.units.equivalencies.spectral` and
-:func:`~astropy.units.equivalencies.spectral_density` (since these return
-lists, they should indeed be combined by adding them together).
-
-The disadvantage of the above approach is that you may forget to turn the
-default off (done by giving an empty argument). To automate this, use
+To enable radians to be treated as a dimensionless unit use
 :func:`~astropy.units.set_enabled_equivalencies` as a `context manager
 <https://docs.python.org/3/reference/datamodel.html#context-managers>`_::
 
@@ -664,5 +650,25 @@ default off (done by giving an empty argument). To automate this, use
   ...    c = np.exp(1j*phase)
   >>> c  # doctest: +FLOAT_CMP
   <Quantity -1.+1.2246468e-16j>
+
+To permanently and globally enable radians to be treated as a dimensionless
+unit use :func:`~astropy.units.set_enabled_equivalencies` not as a context
+manager:
+
+.. doctest-skip::
+
+  >>> u.set_enabled_equivalencies(u.dimensionless_angles())
+  <astropy.units.core._UnitContext object at ...>
+  >>> u.deg.to('')  # doctest: +FLOAT_CMP
+  0.017453292519943295
+
+The disadvantage of the above approach is that you may forget to turn the
+default off (done by giving an empty argument).
+
+:func:`~astropy.units.set_enabled_equivalencies` accepts any list of
+equivalencies, so you could add, for example,
+:func:`~astropy.units.equivalencies.spectral` and
+:func:`~astropy.units.equivalencies.spectral_density` (since these return
+lists, they should indeed be combined by adding them together).
 
 .. EXAMPLE END

